src/main/java/com/limechain/rpc/methods/offchain/OffchainRPCImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.limechain.rpc.methods.offchain;
+
+import com.limechain.runtime.hostapi.dto.InvalidArgumentException;
+import com.limechain.storage.KVRepository;
+import com.limechain.storage.offchain.OffchainStore;
+import org.apache.tomcat.util.buf.HexUtils;
+import org.springframework.stereotype.Service;
+
+@Service
+public class OffchainRPCImpl {
+
+    private static final String PERSISTENT = "PERSISTENT";
+    private static final String LOCAL = "LOCAL";
+    private static final String HEX_PREFIX = "0x";
+    private final OffchainStore persistentStorage;
+    private final OffchainStore localStorage;
+
+    public OffchainRPCImpl(final KVRepository<String, Object> db) {
+        persistentStorage = new OffchainStore(db, true);
+        localStorage = new OffchainStore(db, false);
+    }
+
+    private static byte[] getFromHexString(String key) {
+        return HexUtils.fromHexString(key.substring(2));
```

Comment:
You might want to use `com/limechain/utils/StringUtils::hexToBytes` instead of this method.